### PR TITLE
Add ability to halt deployments after migrations for x time

### PR DESCRIPTION
**Pull request by Dhouti**

### CODE REVIEW

**Comment by Dhouti** [comment]
There's also a fix in here to prevent redis restarts on version change

File: pkg/controller/summon/components/deployment.go [hunk 1]
```diff
@@ -58,7 +58,7 @@ func (comp *deploymentComponent) IsReconcilable(ctx *components.ComponentContext
 	if instance.Status.PostgresStatus != dbv1beta1.StatusReady {
 		return false
 	}
-	if instance.Status.MigrateVersion != instance.Spec.Version {
+	if instance.Status.Status != summonv1beta1.StatusDeploying {
```
Comment (coderanger) [inline pkg/controller/summon/components/deployment.go]: It should also process if in status ready.

File: pkg/controller/summon/components/statefulset.go [hunk 2]
```diff
@@ -55,7 +55,7 @@ func (comp *statefulsetComponent) IsReconcilable(ctx *components.ComponentContex
 	if instance.Status.PostgresStatus != dbv1beta1.StatusReady {
 		return false
 	}
-	if instance.Status.MigrateVersion != instance.Spec.Version {
+	if instance.Status.Status != summonv1beta1.StatusDeploying {
```
Comment (coderanger) [inline pkg/controller/summon/components/statefulset.go]: Ready here too.

File: pkg/controller/summon/components/deployment.go (hunk 1 above)
Comment (Dhouti) [reply pkg/controller/summon/components/deployment.go]: Add StatusReady check to IsReconcilable() && Move StatusDeploying check to beginning of Reconcile()?

**Pull request merged**